### ISSUE TITLE
Some missing error handling in parse_cs

### ISSUE THREAD

**Issue by embray** [astropy/astropy#4158]
In the process of investigating #2970 I ran across what I think is an unrelated buglet.

In my testing I ran
```python
check_conesearch_sites(url_list=None)
```
which seems to go on to download a whole bunch of different catalog resources and look up their conesearch test queries (I'm sort of learning how this all works as I go so please correct me @pllim :)  However, in the process it crashed `parse_cs` because at least *one resource*, given by the ID ivo://CDS.VizieR/J/ApJ/780/34 contains a testQuery element with only an `<sr>` parameter specified, and no `<ra>` or `<dec>`.  Based on my reading of [this](http://www.ivoa.net/documents/REC/DAL/ConeSearch-20080222.html), and some of the discussion [here](http://wiki.ivoa.net/bin/view/IVOA/SimpleDALRegExt10RFC) it seems the RA, DEC, and SR parameters *must* be provided for a valid cone search (duh), though the Resource schema itself doesn't make that entirely clear... :(

In any case I think we should:
a) Report the broken resource and
b) Fix the code to catch this kind of error (the code is currently assuming any testQuery will at least have a valid set of parameters).

**Comment by embray** [comment]
I'll add, this hasn't normally come up in testing since the test uses just a subset of sites for conesearch_master_list, and I was running the full list, I guess.

**Comment by pllim** [comment]
You generally do **not** want to validate *all* the services. I only used that option when I was doing a "first pass" to make a list of "good" services. There are a lot of invalid services in the master list (the last I check, there were over 10k total, good and bad together, mostly bad). That is why I have a short-listed list with the *relatively* "good" ones, and there are functions for users to build their own trusted list.

To account for all possible kinds of errors that the registry can have because the providers did not care to update or correct their info is going to open a can of worms. And I have tried contacting various providers at the start of this project. Most of them did not reply because they don't have resources to fix the problem, or the email got bounced back.

tl;dr - I'd rather mark this as "won't fix".

**Comment by embray** [comment]
@pllim Right, I'm not saying we need to be able to support every possible invalid service; just prevent the code from crashing in this instance.

**Comment by pllim** [comment]
@embray , yes, I agree with you now.
